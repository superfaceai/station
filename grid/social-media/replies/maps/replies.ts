/* eslint-disable jest/no-export */

import { SuperfaceTest } from '@superfaceai/testing';

export const getRepliesTest = (provider: string, postId: string): void => {
  describe(`social-media/replies/${provider}`, () => {
    let superface: SuperfaceTest;

    beforeEach(() => {
      superface = new SuperfaceTest({
        profile: 'social-media/replies',
        provider,
      });
    });

    describe('GetPostComments', () => {
      describe('when a post ID is given', () => {
        it('should list all comments', async () => {
          const result = await superface.run({
            useCase: 'GetPostComments',
            input: { parentId: postId },
          });
          expect(result.isOk()).toBe(true);
          expect(result).toMatchSnapshot();
        });
      });
    });
  });
};
