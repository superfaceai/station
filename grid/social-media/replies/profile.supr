name = "social-media/replies"
version = "1.0.0"

"""
Get Post Replies
Get replies (comments) on a social media post.
"""
usecase GetPostReplies safe {
  input {
    """
    Parent ID
    Identifier of the parent post to get replies for.
    """
    parentId! string!
    """
    Page
    Pagination cursor (obtained from `previousPage` or `nextPage`).
    """
    page string
  }

  result {
    """
    Previous page
    Previous page cursor.
    """
    previousPage string
    """
    Next page
    Next page cursor.
    """
    nextPage string
    """
    Comments
    List of comments under the post.
    """
    comments [Comment]
  }
  
  error {
    """
    Title
    A short, human-readable summary of the problem type.
    """
    title!

    """
    A human-readable explanation specific to this occurrence of the problem.
    """
    detail
  }

  example success {
      input {
          parentId = "be7cdc05-4264-4d47-bc1f-af704d6abf32"
      }

      result {
          comments = [
            {
                id = "123"
                createdAt = "2022-06-09T13:29:11+02:00"
                authorId = "@alberteinstein"
                authorName = "Albert Einstein"
                text = "Hello, World!"
            },
            {
                id = "124"
                createdAt = "2022-06-09T16:29:11+02:00"
                authorId = "@newton"
                authorName = "Isaac Newton"
                lastModifiedAt = "2022-06-09T16:32:42+02:00"
                text = "(deleted.)"
            }
        ]
      }
  }
}

model Comment {
    """
    ID
    Identifier of the reply.
    """
    id! string!

    """
    Created at
    Timestamp of the reply creation in ISO 8601 format.
    """
    createdAt string

    """
    Author ID
    Identificator of the author.
    """
    authorId string

    """
    Author name
    Human-friendly name of the author.
    """
    authorName string

    """
    Last modified
    Timestamp of the last known reply modification in ISO 8601 format.
    """
    lastModifiedAt string

    """
    Text
    Textual content of the comment.
    """
    text string
}