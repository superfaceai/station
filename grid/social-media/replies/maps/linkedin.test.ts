import { getRepliesTest } from './replies';

getRepliesTest('linkedin', 'urn:li:activity:6905096044615405568');
