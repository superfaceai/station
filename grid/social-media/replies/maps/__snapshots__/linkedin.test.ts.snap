// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`social-media/replies/linkedin GetPostComments when a post ID is given should list all comments 1`] = `
Ok {
  "value": Object {
    "comments": Array [
      Object {
        "authorId": "urn:li:organization:81974063",
        "createdAt": undefined,
        "id": "urn:li:comment:(urn:li:activity:6905096044615405568,6930467700934590464)",
        "lastModifiedAt": undefined,
        "text": "rabbit",
      },
      Object {
        "authorId": "urn:li:organization:81974063",
        "createdAt": undefined,
        "id": "urn:li:comment:(urn:li:activity:6905096044615405568,6930466642367774720)",
        "lastModifiedAt": undefined,
        "text": "cats",
      },
    ],
    "nextPage": undefined,
    "previousPage": undefined,
  },
}
`;
